packages/agents/lighthouse/src/tasks/processFromRoot/helpers/xlayer.ts

```diff
@@ -0,0 +1,210 @@
+  amount: string;
+  dest_net: number;
+  dest_addr: string;
+  block_num: string;
+  deposit_cnt: string;
+  network_id: number;
+  tx_hash: string;
+  claim_tx_hash: string;
+  metadata: string;
+  ready_for_claim: boolean;
+  global_index: string;
+};
+
+const getDeposits = async (apiUrl: string, address: string, limit = 100, offset = 0): Promise<Depoist[]> => {
+  try {
+    const res = await axiosGet(`${apiUrl}/bridges/${address}?limit=${limit}&offset=${offset}`);
+    return res.data.deposits !== undefined ? res.data.deposits : [];
+  } catch (e: unknown) {
+    console.log(e);
+    return [];
+  }
+};
+
+const getMerkleProof = async (apiUrl: string, depositCount: string, networkId: number) => {
+  try {
+    const res = await axiosGet(`${apiUrl}/merkle-proof?deposit_cnt=${depositCount}&net_id=${networkId}`);
+    return res.data.proof;
+  } catch (e: unknown) {
+    console.log(e);
+    return null;
+  }
+};
+
+export const getProcessFromXlayerRootArgs = async ({
+  spokeChainId,
+  hubChainId,
+  spokeDomainId,
+  hubDomainId,
+  spokeProvider,
+  hubProvider,
+  message: _message,
+  sendHash: _sendHash,
+  _requestContext,
+}: GetProcessArgsParams): Promise<[]> => {
+  const {
+    logger,
+    config,
+    adapters: { contracts, relayers, chainreader },
+  } = getContext();
+  const { requestContext, methodContext } = createLoggingContext(getProcessFromXlayerRootArgs.name, _requestContext);
+  logger.info("Method start", requestContext, methodContext);
+
+  const xlayerBridgeApiEndpoint =
+    hubChainId === 1 ? "https://rpc.xlayer.tech/priapi/v1/ob/bridge/" : "https://testrpc.x1.tech/priapi/v1/ob/bridge";
+
+  const spokeConnector = contracts.spokeConnector(
+    spokeChainId ?? 0,
+    "Xlayer",
+    config.environment === "staging" ? "Staging" : "",
+  );
+
+  const hubConnector = contracts.hubConnector(
+    hubChainId ?? 0,
+    "Xlayer",
+    config.environment === "staging" ? "Staging" : "",
+  );
+
+  const spokeDeposits = await getDeposits(xlayerBridgeApiEndpoint, spokeConnector!.address);
+  const hubDeposits = await getDeposits(xlayerBridgeApiEndpoint, hubConnector!.address);
+  const claimableDeposits = spokeDeposits
+    .concat(hubDeposits)
+    .filter((d: any) => d.ready_for_claim && d.claim_tx_hash === "");
+
+  if (!claimableDeposits.length) {
+    throw new NoRootAvailable(spokeChainId, hubChainId, requestContext, methodContext, {
+      error: `Claimable desposits don't exist!`,
+    });
+  }
+
+  const hubRpcProvider = new ethers.providers.JsonRpcProvider(hubProvider);
+  const hubConnectorContract = getContract(hubConnector!.address, hubConnector!.abi as any[], hubRpcProvider);
+  const hubAMBAddress: string = await hubConnectorContract.AMB();
+  const spokeRpcProvider = new ethers.providers.JsonRpcProvider(spokeProvider);
+  const spokeConnectorContract = getContract(spokeConnector!.address, spokeConnector!.abi as any[], spokeRpcProvider);
+  const spokeAMBAddress: string = await spokeConnectorContract.AMB();
```

Comment:
note: because amb calls are not going through chain reader they may not have the most robust performance (retries, fallback, etc.)